client/components/contact/ContactForm.jsx

```diff
@@ -0,0 +1,270 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { sendGitRequest } from '@reducers/data';
+import { showFeedbackSuccess, setErrorModal } from '@reducers/ui';
+import { toast } from 'react-toastify';
+
+import 'react-toastify/dist/ReactToastify.css';
+
+import {
+  Container,
+  Grid,
+  Button,
+  TextField,
+  CircularProgress,
+} from '@material-ui/core';
+
+const initialFormValues = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  association: '',
+  message: '',
+  errors: {
+    missingFirstName: false,
+    missingLastName: false,
+    missingEmail: false,
+    invalidEmail: false,
+    missingMessage: false,
+  },
+  loading: false,
+};
+
+const ContactForm = () => {
+  // define the methods to dispatch redux actions
+  const dispatch = useDispatch();
+  const callSendGitRequest = useCallback(obj => dispatch(sendGitRequest(obj)), [dispatch]);
+  const callShowFeedbackSuccess = useCallback(o => dispatch(showFeedbackSuccess(o)), [dispatch]);
+  const callShowErrorModal = useCallback(obj => dispatch(setErrorModal(obj)), [dispatch]);
+
+  // mapStateToProps equivalent
+  const displayFeedbackSuccess = useSelector(state => state.ui.displayFeedbackSuccess);
+  const openErrorModal = useSelector(state => state.ui.error.isOpen);
+
+  const [formValues, setFormValues] = useState(initialFormValues);
+
+  function clearFields() {
+    setFormValues({
+      ...initialFormValues,
+    });
+  }
+
+  // initialize component
+  useEffect(() => {
+    // componentDidMount code goes here...
+    clearFields();
+    if (!!displayFeedbackSuccess === true) {
```

Comment:
i'm using === and !! always guarantees a truthy/falsy value (https://www.codingem.com/javascript-double-exclamation-operator/#:~:text=In%20JavaScript%2C%20the%20double%20exclamation%20operator%20converts%20an,%21%21%29%20is%20actually%20the%20not%20%28%21%29%20operator%20twice.)
in javascript true == 1, but true === 1 is not true (https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons)
using the !! prevents potential edge cases when and if displayFeedbackSuccess does not explicity return "true"

Reply:
I would opt to not use type coercion here since displayFeedbackSuccess should _always_ be a bool, and if it's not, then we have a bug somewhere. It seems like converting to truthy/falsy can hide bugs in this situation, but I'm not sure about the js styles and standards here. Happy to discuss more!

Reply:
updated in latest push